src/workouts/workout.controller.ts

```diff
@@ -1,29 +1,62 @@
-import { Body, Controller, Delete, Get, Param, Post, Put } from "@nestjs/common";
-import { CreateWorkoutDTO } from "./dto/create-workout.dto";
-import { UpdateWorkoutDTO } from "./dto/update-workout.dto";
-import { WorkoutService } from "./workout.service";
+import {
+  Body,
+  Controller,
+  Get,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { CreateWorkoutDTO } from './dto/create-workout.dto';
+import { UpdateWorkoutDTO } from './dto/update-workout.dto';
+import { WorkoutService } from './workout.service';
+import { CurrentUser } from 'src/decorators/currentUser.decorator';
+import { Payload } from 'src/auth/interface/auth.interface';
 
 @Controller('workout')
 export class WorkoutController {
-  constructor (private workoutService: WorkoutService) {}
+  constructor(private workoutService: WorkoutService) { }
 
   @Post()
-  async createWorkout(@Body() workoutPayload: CreateWorkoutDTO) {
-    return this.workoutService.createWorkout(workoutPayload);
+  async createWorkout(
+    @CurrentUser() userId: number,
+    @Body() body: CreateWorkoutDTO,
+  ) {
+    return this.workoutService.createWorkout(body, userId);
   }
 
   @Get()
-  async getAllWorkouts() {
-    return this.workoutService.workouts({});
+  async getAllWorkouts(@CurrentUser() user: Payload) {
+    return this.workoutService.workouts({
+      where: {
+        userId: user.sub
+      },
+    });
   }
 
   @Get(':id')
-  async getOneWorkout(@Param('id') id: string) {
-    return this.workoutService.workout({ id: Number(id) });
+  async getOneWorkout(@Param('id') id: string, @CurrentUser() userId: number) {
+    return this.workoutService.workout({
+      id: Number(id),
+      AND: {
+        userId: userId
+      }
+    });
```

Comment:
Nesses lugares aqui cê dá deixando a lógica do Prisma no controller, pra mim não faz sentido. Já que o prisma tá só dentro do service, eu mandaria `workoutId` e `userId` pra dentro do service, e _**lá dentro**_ eu faria o `AND`. Vale o mesmo pros `where`s kkkkkkkk